accounts-db/src/accounts_db.rs

```diff
@@ -11110,39 +11138,67 @@ pub mod tests {
                 (false, ())
             });
 
-            let zero_lamport_single_ref_pubkeys = [&pubkey_zero];
+            let zero_lamport_single_ref_pubkeys =
+                if pass < 2 { vec![&pubkey_zero] } else { vec![] };
```

Comment:
hmm. this was previously an illegal call to `remove_zero_lamport_single_ref_accounts_after_shrink` when there was a newer alive account. Probably fine to modify test like this.